processor/workers.go

```diff
@@ -137,18 +148,18 @@ func codeState(
 	endComments [][]byte,
 	langFeatures LanguageFeature,
 	digest *hash.Hash,
-) (int, int64, []byte, [][]byte) {
+) (int, int64, []byte, [][]byte, bool) {
```

Comment:
This list of return values is getting a little long and opaque. Given that it's a longer function, it took me a bit of scanning back and forth to figure out what it's actually for. It might be time to think about making an explicit struct for the return value, or at least using named return values.

Reply:
Yep I 100% want this, and am working on it in a new branch.

Reply:
BTW I have this on another branch. If we can get this in first refactoring this to be easier to understand is my first priority. I found it bloody annoying to work with while dealing with it as well.